staging/src/k8s.io/component-base/logs/options.go

```diff
@@ -63,13 +63,16 @@ func (o *Options) AddFlags(fs *pflag.FlagSet) {
 	unsupportedFlags := fmt.Sprintf("--%s", strings.Join(unsupportedLoggingFlags(), ", --"))
 	formats := fmt.Sprintf(`"%s"`, strings.Join(logRegistry.List(), `", "`))
 	fs.StringVar(&o.LogFormat, logFormatFlagName, defaultLogFormat, fmt.Sprintf("Sets the log format. Permitted formats: %s.\nNon-default formats don't honor these flags: %s.\nNon-default choices are currently alpha and subject to change without warning.", formats, unsupportedFlags))
+
+	// Frezze Log Registry after setting logger
+	// which make it unable to register new logger
+	logRegistry.freezed = true
```

Comment:
All other functions used from registry are public, can you add function `Freeze` and call it here?